src/syscall_handler.c

```diff
@@ -86,6 +87,27 @@ int sys_run( const char *path )
     for(i=0;i<p->window_count;i++) {
         p->windows[i]->count++;
     }
+
+    /* Copy command line arguments */
+	struct x86_stack *s = (struct x86_stack *) p->stack_ptr;
+    unsigned paddr;
+    pagetable_getmap(p->pagetable,PROCESS_STACK_INIT+0xF-PAGE_SIZE+1,&paddr);
+    char* esp = (char*)paddr+PAGE_SIZE-0x10;
+    char* ebp = esp;
+    /* Copy each argument */
+    for (i = 0; i < argc; i++) {
+        ebp -= 256;
+        strncpy(ebp, argv[i], 255);
```

Comment:
It's fine to have some reasonable upper limits for these strings, but use a macro instead of a magic number, in case they need to change..